core/src/main/java/nu/yona/server/analysis/entities/WeekActivity.java

```diff
@@ -84,11 +82,13 @@ public void addDayActivity(DayActivity dayActivity)
 			throw new IllegalStateException("Week is already full (" + dayActivities.size() + " days present)");
 		}
 
+		dayActivity.setWeekActivity(this);
 		dayActivities.add(dayActivity);
 	}
 
 	public void removeAllDayActivities()
 	{
+		dayActivities.forEach(da -> da.setWeekActivity(null));
```

Comment:
Indeed, the deletion is taken care of through orphan removal. I believe I had an issue calling delete directly, but I'm not entirely sure. Do you believe it's worthwhile retesting that? I expect it to be identical in performance.